### PR TITLE
enable nullable for commands

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Review by manfred-brands — CHANGES_REQUESTED** [review]
The original design of the protected field that needs to be set in derived classes is not the best and can be done better, either using a abstract method or a constructor parameter.

File: src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs
Comment (SimonCropp) [reply src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs:56]: wouldnt that be a breaking change?

Comment (manfred-brands) [reply src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs:56]: Yes and even though the classes are in an internal namespace they are still public.

So let stick to exisiting design and simply declare the Action fields as nullable. They are already checked for null before use.

Comment (SimonCropp) [reply src/NUnitFramework/framework/Internal/Commands/BeforeAndAfterTestCommand.cs]: done

**Comment by SimonCropp** [comment]
this one should be good to go

**Comment by SimonCropp** [comment]
@manfred-brands 

> The original design of the protected field that needs to be set in derived classes is not the best and can be done better, either using a abstract method or a constructor parameter.

i would prefer to do that in another PR

**Comment by SimonCropp** [comment]
bump

File: src/NUnitFramework/framework/Internal/Commands/SetUpTearDownCommand.cs
Comment (stevenaw) [inline src/NUnitFramework/framework/Internal/Commands/SetUpTearDownCommand.cs:25]: 👍 for this simplification

File: src/NUnitFramework/framework/Internal/Commands/ConstructFixtureCommand.cs
Comment (stevenaw) [inline src/NUnitFramework/framework/Internal/Commands/ConstructFixtureCommand.cs:30]: It took me a bit to think through the possibilities here to ensure this was still the same logic but it seems sound

**Review by stevenaw — APPROVED** [review]
Thank you for this @SimonCropp !
LGTM but I'd appreciate a second pair of eyes for review before merging.

**Comment by SimonCropp** [comment]
@stevenaw who would be best to do that?

**Comment by stevenaw** [comment]
Oh, I see @manfred-brands already had a look.

**Pull request merged**
